regression/test.pl

```diff
@@ -117,10 +113,27 @@ ($$$$$)
           $included--;
         } else {
           my $r;
-          $result =~ s/\\/\\\\/g;
-          $result =~ s/([^\\])\$/$1\\r\\\\?\$/;
-          system("bash", "-c", "grep $grep_options \$'$result' '$name/$output' >/dev/null");
-          $r = ($included ? $? != 0 : $? == 0);
+
+          # Multi line therefore we run each check across the whole output
+          if($grep_options eq "activate-multi-line-match") {
+            my $dir = getcwd();
+            my $abs_path = "$dir/$name/$output";
+            open( my $fh => $abs_path) || die "Cannot open '$name/$output': $!";
+
+            local $/ = undef;
+            binmode $fh;
+            my $whole_file = <$fh>;
+            my $is_match = $whole_file =~ /$result/;
+            $r = ($included ? !$is_match : $is_match);
+            close($fh);
+          }
+          else
+          {
+            $result =~ s/\\/\\\\/g;
+            $result =~ s/([^\\])\$/$1\\r\\\\?\$/;
+            system("bash", "-c", "grep \$'$result' '$name/$output' >/dev/null");
```

Comment:
Yeah sadly a bunch fail since of escaping characters https://github.com/diffblue/cbmc/pull/505#discussion_r100527106 I can find out exactly which ones if you want but it was more than one (or I would have just fixed it :P)